vmaas/reposcan/database/csaf_store.py

```diff
@@ -28,151 +56,327 @@ def delete_csaf_file(self, name: str):
-                        INSERT INTO csaf_products (cpe, package, module)
-                        VALUES %s returning id;
-                        """, (products), fetch=True)
-                    inserted_products.extend([(result[0], status) for result in ids])
-                self.conn.commit()
+            rows = execute_values(
+                cur,
+                """
+                    INSERT INTO csaf_product (cpe_id, package_name_id, package_id, module_stream_id)
+                    VALUES %s
+                    RETURNING id, cpe_id, package_name_id, package_id, module_stream_id
+                """,
+                products.to_tuples(
+                    ("cpe_id", "package_name_id", "package_id", "module_stream_id"),
+                    missing_only=True,
+                    with_cpe_id=True,
+                    with_pkg_id=True,
+                ),
+                fetch=True,
+            )
+            self.conn.commit()
+            self._set_product_ids(rows, products)
         except Exception as exc:
             CSAF_FAILED_IMPORT.inc()
-            self.logger.exception("Failure while inserting data into csaf_products: %s", exc)
+            self.logger.exception("Failure while inserting data into csaf_product: %s", exc)
             self.conn.rollback()
+            raise CsafStoreException("Failure while inserting data into csaf_product")
         finally:
             cur.close()
 
-        return inserted_products
+    def _insert_cves(self, cve: str, products: model.CsafProducts) -> None:
+        if not products:
+            raise CsafStoreSkippedCVE("cannot map any products")
 
-    def _insert_cves(self, cve, products):
+        cur = self.conn.cursor()
         try:
-            cur = self.conn.cursor()
-            if products:
-                execute_values(cur, """
-                    INSERT INTO csaf_cves (cve, csaf_product_id, product_status_id)
+            cur.execute("SELECT id FROM cve WHERE UPPER(name) = %s", (cve,))
+            if (row := cur.fetchone()) is None:
+                raise CsafStoreException(f"{cve} not found in DB")
+
+            file_id = self.cve2file_id[cve]
+            to_upsert = [
+                (row[0], id_, status_id, file_id)
+                for id_, status_id in products.to_tuples(("id_", "status_id"), with_id=True)
+            ]
+            execute_values(
+                cur,
+                """
+                    INSERT INTO csaf_cve_product (cve_id, csaf_product_id, csaf_product_status_id, csaf_file_id)
                     VALUES %s
-                    ON CONFLICT (cve, csaf_product_id)
-                    DO UPDATE SET product_status_id = EXCLUDED.product_status_id;
-                    """, [(cve, product[0], product[1]) for product in products])
-                self.conn.commit()
+                    ON CONFLICT (cve_id, csaf_product_id)
+                    DO UPDATE SET csaf_product_id = EXCLUDED.csaf_product_id
```

Comment:
It was previously `DO UPDATE SET product_status_id = EXCLUDED.product_status_id`, is this an error?

Reply:
yes, that's an error, thanks